ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/KafkaConfigStoreTest.java

```diff
@@ -0,0 +1,204 @@
+          )
+      );
+    }
+    when(consumer.poll(any())).thenReturn(
+        new ConsumerRecords<>(
+            Collections.singletonMap(
+                topicPartition,
+                records
+            )
+        )
+    );
+    return consumer;
+  }
+
+  private void verifyDrainLog(
+      final KafkaConsumer<String, KsqlProperties> consumer,
+      final InOrder inOrder,
+      final int nmsgs) {
+    inOrder.verify(consumer).assign(topicPartitionAsList);
+    inOrder.verify(consumer).seekToBeginning(topicPartitionAsList);
+    for (int i = 0; i < nmsgs; i++) {
+      inOrder.verify(consumer).poll(any());
+    }
+  }
+
+  @Test
+  public void shouldWriteConfigIfNoConfigWritten() {
+    // Given:
+    final KafkaConsumer<String, KsqlProperties> consumerBefore = expectRead();
+    final KafkaConsumer<String, KsqlProperties> consumerAfter = expectRead(
+        new KafkaConfigStore.KsqlProperties(
+            ksqlConfig.getAllConfigPropsWithSecretsObfuscated()));
+    when(consumerSupplier.get()).thenReturn(consumerBefore).thenReturn(consumerAfter);
+
+    // When:
+    initConfigStore();
+    final KsqlConfig resolvedConfig = configStore.getKsqlConfig();
+
+    // Then:
+    assertThat(resolvedConfig.values(), equalTo(ksqlConfig.values()));
+    assertThat(
+        resolvedConfig.getKsqlStreamConfigProps(),
+        equalTo(ksqlConfig.getKsqlStreamConfigProps()));
+    final InOrder inOrder = Mockito.inOrder(consumerBefore, consumerAfter, producer);
+    verifyDrainLog(consumerBefore, inOrder, 0);
+    @SuppressWarnings("unchecked")
+    final ArgumentCaptor<ProducerRecord<String, KafkaConfigStore.KsqlProperties>> msgCaptor
+        = ArgumentCaptor.forClass(ProducerRecord.class);
+    inOrder.verify(producer).send(msgCaptor.capture());
+    assertThat(
+        msgCaptor.getValue().value().getKsqlProperties(),
+        equalTo(ksqlConfig.getAllConfigPropsWithSecretsObfuscated()));
+    inOrder.verify(producer).flush();
+    verifyDrainLog(consumerAfter, inOrder, 1);
+  }
+
+  @Test
+  public void shouldReadAndMergeExistingConfigIfExists() {
+    // Given:
+    final KsqlConfig persistedConfig = new KsqlConfig(
+        Collections.singletonMap(
+            KsqlConfig.KSQL_PERSISTENT_QUERY_NAME_PREFIX_CONFIG, "not-the-default"
+        )
+    );
+    final KafkaConsumer<String, KsqlProperties> consumer = expectRead(
+        new KafkaConfigStore.KsqlProperties(
+            persistedConfig.getAllConfigPropsWithSecretsObfuscated()));
+    when(consumerSupplier.get()).thenReturn(consumer);
+
+    // When:
+    initConfigStore();
+    final KsqlConfig resolvedConfig = configStore.getKsqlConfig();
+
+    // Then:
+    assertThat(
+        resolvedConfig.getString(KsqlConfig.KSQL_PERSISTENT_QUERY_NAME_PREFIX_CONFIG),
+        equalTo("not-the-default"));
+    verifyZeroInteractions(producer);
+  }
+
+  @Test
+  @SuppressFBWarnings("DM_DEFAULT_ENCODING")
+  public void shouldDeserializeMissingContentsToNull() {
```

Comment:
name seems out of alignment with test.  Test looks to deserialise to empty map, not null.